cmd/thanos/compact.go

```diff
@@ -242,10 +247,13 @@ func runCompact(
 
 	duplicateBlocksFilter := block.NewDeduplicateFilter()
```

Comment:
Somewhere we need to enable vertical compaction flag IF replica flag is enabled on LeveledCompactor